strapi/strapi
# feat(content-releases): add publish webhook

### What does it do?

- Adds a new webhook `releases.publish` to trigger when a publish is attempted to provide feedback to the user.
- Changes the logic of the Webhook Events table to display multiple tables for different features.
- Emits event when publishing a release to capture them and trigger the webhook.

## Testing

### Enabling the future flags

The scheduling functionality is implemented through future flags. To begin, ensure the future flag is set to true. Navigate to `config/features.js` and include `contentReleasesScheduling` as an option within your future object. Create the future object if it does not exist.

Example:
```js
// config/features.js

module.exports = ({ env }) => ({
  future: {
    contentReleasesScheduling: true,
  },
});

```

### How to test it

1. You should setup a server locally to listen to `POST` calls on the endpoint and port you want to completely test this feature.
2. When you have your server running locally, go to Settings -> Webhooks and add a new webhook with for the "Releases: Publish" event using your server url.
3. When publishing an event it should trigger the webhook
